test/e2e/tests/http3.go

```diff
@@ -0,0 +1,220 @@
+			req.Server = server
+			cReq, cResp, err := CaptureRoundTrip(req, gwAddr)
+			if err != nil {
+				t.Errorf("failed to get expected response: %v", err)
+			}
+
+			if err := http.CompareRequest(t, &req, cReq, cResp, expectedResponse); err != nil {
+				t.Errorf("failed to compare request and response: %v", err)
+			}
+		})
+	},
+}
+
+func CaptureRoundTrip(request roundtripper.Request, gwAddr string) (*roundtripper.CapturedRequest, *roundtripper.CapturedResponse, error) {
+	transport, err := httpTransport(request, gwAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+	return defaultRoundTrip(request, transport)
+}
+
+func httpTransport(request roundtripper.Request, gwAddr string) (*http3.RoundTripper, error) {
+
+	transport := &http3.RoundTripper{
+		QuicConfig: &quic.Config{
+			Tracer: qlog.DefaultTracer,
+		},
+		Dial: func(ctx context.Context, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlyConnection, error) {
+			if addr == "www.example.com:443" {
+				return quic.DialAddrEarly(ctx, gwAddr, tlsCfg, cfg)
+			}
+			return nil, nil
+		},
+	}
+	if request.Server != "" && len(request.CertPem) != 0 {
+		tlsConfig, err := tlsClientConfig(request.Server, request.CertPem)
+		if err != nil {
+			return nil, err
+		}
+		transport.TLSClientConfig = tlsConfig
+	}
+	return transport, nil
+}
+
+func tlsClientConfig(server string, certPem []byte) (*tls.Config, error) {
+	// Add the provided cert as a trusted CA
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(certPem) {
+		return nil, fmt.Errorf("unexpected error adding trusted CA")
+	}
+
+	if server == "" {
+		return nil, fmt.Errorf("unexpected error, server name required for TLS")
+	}
+
+	// Create the tls Config for this provided host, cert, and trusted CA
+	// Disable G402: TLS MinVersion too low. (gosec)
+	// #nosec G402
+	return &tls.Config{
+		ServerName: server,
+		RootCAs:    certPool,
+	}, nil
+}
+
+func defaultRoundTrip(request roundtripper.Request, transport *http3.RoundTripper) (*roundtripper.CapturedRequest, *roundtripper.CapturedResponse, error) {
```

Comment:
can you make all the name of these functions explicit with `http3`? may help others to know these functions are exclusive for HTTP3